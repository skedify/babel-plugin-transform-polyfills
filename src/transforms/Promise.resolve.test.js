describe('Promise.resolve', () => {
  it('should replace Promise.resolve with a polyfill', () => {
    const result = compile(
      `
      const c = Promise.resolve()
    `
    )

    return expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Promise.resolve(Promise.resolve([1, 2]))
    `
    )

    return expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Promise.resolve())
    `
    )

    return expect(result).toMatchSnapshot()
  })
})
