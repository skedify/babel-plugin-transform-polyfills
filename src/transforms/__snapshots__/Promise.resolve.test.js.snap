// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise.resolve should not import polyfill more than once 1`] = `
"import _Promise_resolve from \\"babel-plugin-transform-polyfills/polyfills/Promise/resolve\\";

const e = _Promise_resolve(_Promise_resolve([1, 2]));"
`;

exports[`Promise.resolve should replace Promise.resolve with a polyfill 1`] = `
"import _Promise_resolve from \\"babel-plugin-transform-polyfills/polyfills/Promise/resolve\\";

const c = _Promise_resolve();"
`;

exports[`Promise.resolve should work when part of a call chain 1`] = `
"import _Promise_resolve from \\"babel-plugin-transform-polyfills/polyfills/Promise/resolve\\";

const c = Object.keys(_Promise_resolve());"
`;
